### ISSUE TITLE
Idea: warn about repeated expressions

### ISSUE THREAD

**Issue by w0rp** [palantir/tslint#3152]
Here is an idea for a new rule, which requires type information and can be quite complex. TSLint could warn about redundant conditions, which could reveal typos in code. Consider the following code.

```ts
if (x > 0 && x > 0) {
}
```

The check for `x > 0` is redundant in the above code, and you might have wanted to write `y > 0` instead, but wrote `x` by mistake.

I'm mainly writing this down as I might try implementing this myself in the future.

**Comment by ajafff** [comment]
Some random thoughts:

Does this rule only look for exact duplicates?
```ts
if (x >= 0 && x > 0) {} // error here too? note the >=
```
It could/should check following `else if` conditions:
```ts
if (x > 0) {
} else if (x > 0) { // error on this condition
}
```
... and maybe even nested if-statements:
```ts
if (x > 0) {
    if (x > 0) { // error on this condition
    }
}
```
... and how about control flow analysis?
```ts
if (x > 0) {
return;
} 
if (x > 0) { // error on this condition, because it is always false
}
```

Also note that this would be way easier to implement if typescript had type ranges https://github.com/Microsoft/TypeScript/issues/15480 and subtraction types https://github.com/Microsoft/TypeScript/issues/4183

**Comment by w0rp** [comment]
The way I see it, there is just about no end to how far you could take this idea, so I'd start with looking for the same expressions for variables and attributes inside one local Boolean expression.

**Comment by JoshuaKGoldberg** [comment]
Note: per #4534, this issue will be closed in less than a month if no PR is sent to add the rule. If you _really_ need the rule, custom rules are always an option and can be maintained outside this repo!
